packages/developer-portal/src/components/pages/webhooks/webhook-ping-form.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { Dispatch, FC, SetStateAction, useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Dispatch as ReduxDispatch } from 'redux'
+import { updateWebhookCreateEditState } from '../../../actions/webhooks-subscriptions'
+import { selectWebhookSubscriptionTopics } from '../../../selector/webhooks-subscriptions'
+import { PingWebhooksByIdParams, WebhookModel } from '../../../services/webhooks'
+import { developerSetWebhookPingStatus, developerWebhookPing } from '../../../actions/developer'
+import Routes from '../../../constants/routes'
+import {
+  ExpandableContentSize,
+  elP8,
+  Subtitle,
+  elMl3,
+  InputGroup,
+  Button,
+  Label,
+  ColSplit,
+  Grid,
+  ButtonGroup,
+  elMlAuto,
+  Select,
+  BodyText,
+  useSnack,
+  InputAddOn,
+  elMb6,
+  useModal,
+  PersistantNotification,
+  FlexContainer,
+} from '@reapit/elements'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useForm } from 'react-hook-form'
+import { object, string } from 'yup'
+import errorMessages from '../../../constants/error-messages'
+import { handleCollapseRow } from './webhooks-manage-form'
+import { WebhookCreateEditState } from '../../../reducers/webhooks-subscriptions/webhook-edit-modal'
+import { link } from '../../../styles/elements/link'
+import { openNewPage } from '../../../utils/navigation'
+import { selectWebhookTestStatus } from '../../../selector'
+import { WebhookPingTestStatus } from '../../../reducers/developer'
+
+interface WebhooksPingFormProps {
+  webhookModel: WebhookModel
+  setIndexExpandedRow: Dispatch<SetStateAction<number | null>>
+  setExpandableContentSize: Dispatch<SetStateAction<ExpandableContentSize>>
+}
+
+export const handlePingWebhook =
+  (webhookModel: WebhookModel, dispatch: ReduxDispatch, setWebhookPingId: Dispatch<SetStateAction<string | null>>) =>
+  (values: PingWebhookFormSchema) => {
+    const { topicId } = values
+    const { id: webhookId } = webhookModel
+    if (!webhookId || !topicId) return
```

Comment:
Throw exception or create notification?

Reply:
In practice, the form is validated - you will never hit this line because the form won't submit. It's just to make TS play nice

Reply:
ah fair :)